### ISSUE TITLE
Better sidebar markup: distinguish content with and without sub sections.

### ISSUE THREAD

**Issue by trusktr** [docsifyjs/docsify#1181]
I've been working on my site, gathering ideas.

## Feature request

Improvement of markup so that we know if a menu item will have sub-menu items.

#### What problem does this feature solve?

Currently, Docsify assumes that every top-level menu item must have sub-content, so therefore `docsify-themeable` styles the meny items with the open/close icons. There's no way for docsify-themeable to know if a menu item should not be open/close-able.

#### What does the proposed API look like?

Somehow, the markup should convey whether the menu items will be collapsible or not. This would perhaps add classes in the markup.

#### How should this be implemented in your opinion?

There's no way for the client-side engine to know ahead of time whether a page will have sub-sections, or not.

Perhaps the only way to do this currently, would be to provide a configuration option, like

```js
window.$docsify = {
  nonCollapsibleMenuItems: [
    './foo/m.md',
    'google.com',
  ]
}
```

Or maybe, we can expand on the sidebar markdown parser to detect a certain syntax (like Docsify does with other links) to tell it which ones are not collapsible:

```md
- [Foo](./foo.md ':no-collapse') 
```

I think I like the markdown idea more.

This is what my sidebar markdown looks like:

```md
-   **Guide**

    -   [Install](/install.md)
    -   [Workflows](/workflows.md)

-   **Examples**

    -   [Hello 3D](/examples/hello3d.md)
    -   [Hello 3D, Parent Transforms](/examples/hello3d-parent-transforms.md)
    -   [Ripple flip](/examples/ripple-flip.md)
    -   [Autolayout (declarative)](/examples/autolayout-declarative.md)
    -   [Autolayout (imperative)](/examples/autolayout-imperative.md)
    -   [Morphing spiral](/examples/spiral.md)
    -   [obj-model](/examples/obj-model.md)
    -   [Buttons with shadow](/examples/buttons-with-shadow.md)
    -   [material-texture](/examples/material-texture.md)
    -   [Origin](/examples/origin.md)

-   **API Reference**
    <!-- __API_AUTOGENERATED_BEGIN__ -->
  - components/
    - [Cube](/api/components/Cube.md)
  - core/
    - [ImperativeBase](/api/core/ImperativeBase.md)
    - [Node](/api/core/Node.md)
    - [TreeNode](/api/core/TreeNode.md)
  - html/
    - [WithUpdate](/api/html/WithUpdate.md)

<!-- __API_AUTOGENERATED_END__ -->
```

And this is currently what it looks like (work in progress):

![current-sidebar](https://user-images.githubusercontent.com/297678/82156546-73fc7b00-9830-11ea-8199-e2db806eccc1.png)

The items under `Examples` are all non-collapsible, so I'd like to remove the collapse icons from those. They are the docsify-themeable icons (@jhildenbiddle).

What are your thoughts on how to handle this?

**Comment by jhildenbiddle** [comment]
Makes sense.

Instead of specifying the collapsable yes/no, perhaps we should allow people to specify the number of sub levels a menu link should display (similar to the [subMaxLevel](https://docsify.js.org/#/configuration?id=submaxlevel), but that is a global setting for all sidebar links). Setting something like `:sub-level=0` would be the equivalent and setting `:no-collapse`, while setting it a number greater than `0` would limit the sub levels rendered when the sidebar section is expanded to that number.

**Comment by trusktr** [comment]
Yeah, I think some sort of helper would make sense, especially since Docsify currently can't tell.

For now, I think I can workaround the issue by setting a class , like `[Foo](./foo.md ':class=remove-chevron')`. I'm gonna give this a shot.

**Comment by trusktr** [comment]
Hmm, actually that's not working for me, `[Foo](./foo.md ':class=remove-chevron')` results in an `<a>` tag with no `class=""` applied to it in both the sidebar and main content area.

**Comment by anikethsaha** [comment]
I like the `:sub-level` option as this can give more details.

> Hmm, actually that's not working for me, `[Foo](./foo.md ':class=remove-chevron')` results in an `<a>` tag with no `class=""` applied to it in both the sidebar and main content area.

I think this feature is not available for sidebars,


@trusktr this will be for v5 right ?

**Comment by trusktr** [comment]
> I think this feature is not available for sidebars,

True, as most features operate on the content area only. However with this particular feature, I tried it in my markdown pages, and also could not get it to work there. It might be abug.

**Comment by vagra** [comment]
see my demo site: 
https://kepan.org/alpha/#/kepan/

is it you want?

**Comment by vagra** [comment]
I write `has-children` class for `li` in tpl.js:

```
export function tree(toc, tpl = '<ul class="app-sub-sidebar">{inner}</ul>') {
  if (!toc || !toc.length) {
    return '';
  }

  let innerHTML = '';
  toc.forEach(node => {
    if (node.children) {
      innerHTML += `<li class="has-children">`;
      innerHTML += `<a class="section-link" href="${node.slug}">${node.title}</a>`;
      innerHTML += tree(node.children, tpl);
      innerHTML += `</li>`;
    } else {
      innerHTML += `<li><a class="section-link" href="${node.slug}">${node.title}</a></li>`;
    }
  });
  return tpl.replace('{inner}', innerHTML);
}
```

**Comment by vagra** [comment]
hello.

I make some changes for sidebar item:
https://github.com/vagra/docsify/commit/56e643a45f18db31c2995acee5a8a8ca4b0459cd

changes:

1. when expand, display down arrow, when collapse, display right arrow;

2. this is only for `app-sub-sidebar`;

3. if `loadSidebar` set to true, the sidebar level is:
  - `sidebar-nav` for _sidebar.md 
  - `app-sub-sidebar` for main content

so, when `loadSidebar` set to true, the sidebar has two level: 
  - `sidebar-nav`
    - `app-sub-sidebar`
https://kepan.org/side1/

when `loadSidebar` set to false, the sidebar has only one level:
  - `app-sub-sidebar`
https://kepan.org/side0/

this making the tree of main content can expand/collapse always.

**Comment by vagra** [comment]
prev method only change icons and add expand/collpse function for `app-sub-sidebar`, not for `sidebar-nav`.

**Comment by jhildenbiddle** [comment]
This issue has been addressed by PR #2469. The changes will be available when Docsify v5 is released.
